### PR TITLE
Step-9: Add toast message notification

### PR DESCRIPTION
**Pull request by sta-wtag**
### **[Step-9](https://trello.com/c/Ke8ow7KL/11-step-9-add-toast-message-notification)**

Add toast message notification system

### **Changes**
- [x] Show toast message when a card is added.
- [x] Show a toast message when a card is deleted.
- [x] Show a toast message when something goes wrong.

### CODE REVIEW

File: store/todos.js [hunk 1]
```diff
@@ -143,6 +146,7 @@ export const mutations = {
     state.totalPage = Math.ceil(state.taskListPerPage.length / state.limit);
   },
   deleteTask: (state, val) => {
+    console.log(val.id);
```
Comment (fza-wtag) [inline store/todos.js:149]: Apu, remove this console.log if not needed.

Comment (fza-wtag) [inline store/todos.js:149]: Apu, remove this console log if not needed.

**Review by sai-wtag — APPROVED** [review]
We can add a cross (X) button, to manually remove the toast.
Other than that, LGTM. Thanks

image

File: assets/css/main.css [hunk 2]
```diff
@@ -151,3 +151,41 @@ textarea {
...
+}
+.error {
+  position:absolute;
+  height: 55px;
+  background-color: #E85F5F;
+  top:90px;
+  left:0px;
+  right:0px;
+  border-radius: 5px;
+}
```
Comment (hhu-wtag) [inline assets/css/main.css]: `.success` and `.error` are sharing many classes. Can we create a `.swal-basic` class and reuse properties?

File: components/TaskCard.vue [hunk 3]
```diff
@@ -164,6 +176,14 @@ export default {
...
       this.showEditIcon = true;
       this.loading = false;
+      this.alert();
+    },
+    alert() {
+      swal(this.$t('alert.message.success'), {
+        buttons: false,
+        className: 'success',
+        timer: 3000,
+      });
```
Comment (hhu-wtag) [inline components/TaskCard.vue]: Let's move it into `utils.js` and reuse it.

File: components/TaskContainer.vue [hunk 4]
```diff
@@ -134,6 +135,12 @@ export default {
       if (this.taskDescription.trim().length <= 0) {
         this.titleInputError = true;
         this.titleErrorMsg = 'Field is empty';
+        swal(this.$t('alert.message.error'), {
+          buttons: false,
+          className: 'error',
+          iconHtml: '<img src="https://picsum.photos/100/100">',
+          timer: 3000,
+        });
```
Comment (hhu-wtag) [inline components/TaskContainer.vue]: ditto

File: components/TaskContainer.vue [hunk 5]
```diff
@@ -143,6 +150,12 @@ export default {
     addTask() {
       this.$store.dispatch('todos/addTask', this.taskDescription);
       this.$store.dispatch('todos/setTotalPage');
+      swal(this.$t('alert.message.success'), {
+        buttons: false,
+        className: 'success',
+        iconHtml: '<img src="https://picsum.photos/100/100">',
+        timer: 3000,
+      });
```
Comment (hhu-wtag) [inline components/TaskContainer.vue]: ditto

File: store/todos.js [hunk 6]
```diff
@@ -158,6 +159,7 @@ export const mutations = {
...
```
Comment (hhu-wtag) [inline store/todos.js]: Can we remove this if not needed?

File: tests/unit/TaskCard.spec.js [hunk 7]
```diff
@@ -1,18 +1,22 @@
 import { mount } from '@vue/test-utils';
-import Nuxti18n from '@nuxtjs/i18n';
+// import Nuxti18n from '@nuxtjs/i18n';
```
Comment (hhu-wtag) [inline tests/unit/TaskCard.spec.js]: ditto

**Review by hhu-wtag — CHANGES_REQUESTED** [review]
Some small changes requested. Nice work overall.

File: assets/css/main.css (hunk 2 above)
Comment (sta-wtag) [reply assets/css/main.css:171]: Resolved

File: store/todos.js (hunk 6 above)
Comment (sta-wtag) [reply store/todos.js]: Resolved

File: tests/unit/TaskCard.spec.js (hunk 7 above)
Comment (sta-wtag) [reply tests/unit/TaskCard.spec.js]: Resolved

File: components/TaskCard.vue (hunk 3 above)
Comment (sta-wtag) [reply components/TaskCard.vue]: resolved

File: assets/css/main.css (hunk 2 above)
Comment (sta-wtag) [reply assets/css/main.css]: Swal className property is not taking two classNames for some reason such as 'swal-basic success' is throwing DOM error as there can not be spaces between class names and ['swal-basic', 'success'] is not applying any style to the element

File: components/TaskContainer.vue (hunk 4 above)
Comment (sta-wtag) [reply components/TaskContainer.vue]: resolved

File: components/TaskContainer.vue (hunk 5 above)
Comment (sta-wtag) [reply components/TaskContainer.vue]: resolved

**Review by hhu-wtag — APPROVED** [review]

**Review by bia-wtag — APPROVED** [review]
Tested code changes, LGTM. I thin there is an empty file added here, could we remove it? 
https://github.com/sta-wtag/TodoApp/blob/f2b7a182f6421a915c3413861b837e0ef3c797e5/components/ToastComponent

**Pull request merged**